Rebase on the Kubernetes images

googlielmo opened the pull request:
To be merged only after https://github.com/hazelcast/hazelcast-docker/pull/41 and after publishing the corresponding images on Docker Hub

googlielmo commented:
Thank you so much @lazerion !

googlielmo commented on hazelcast-enterprise-openshift-centos/Dockerfile:
```diff
@@ -1,21 +1,9 @@
-ENV HZ_CP_MOUNT ${HZ_HOME}/external
-ENV JAVA_HOME /usr/lib/jvm/java
-ENV LANG en_US.utf8
+ENV HZ_VERSION 3.9
```
Actually, the `HZ_VERSION` is defined in the base image, and available to the label.
I think just having `latest` in the FROM image is sufficient when building the latest released version. 
@hasancelik

hasancelik replied:
@mesutcelik @googlielmo If we do it like this -using `latest`-, release script update is not needed :+1:

hasancelik commented on hazelcast-enterprise-openshift-centos/Dockerfile:
```diff
@@ -1,21 +1,9 @@
-ARG HZ_KUBE_VERSION=1.0.0
-ARG HZ_MAVEN_DIR=${HZ_VERSION}
-ARG REPOSITORY_URL=https://repository-hazelcast-l337.forge.cloudbees.com
+FROM hazelcast/hazelcast-enterprise-kubernetes:${HZ_IMAGE_TAG}
```
👍

googlielmo commented:
I've addressed comments @mesutcelik 
Using `latest` image tag at release time. 
For testing it can be overridden with `--build-arg` at build time.
Thanks @hasancelik

mesutcelik commented on hazelcast-enterprise-openshift-centos/Dockerfile:
```diff
@@ -1,21 +1,9 @@
-
-ENV USER_NAME=hazelcast
-ENV USER_UID=10001
+ARG HZ_IMAGE_TAG=latest
```
I am not sure using latest here is a good choice because github tag will not include exact released version.

googlielmo commented:
@mesutcelik I've addressed your comments and reverted to explicit version number.
This will allow one to rebuild the exact same image from a tag.
cc @hasancelik

googlielmo commented:
I've updated the READMEs to mention the base images.

googlielmo commented:
I think we can merge this PR, thank you guys for the comments.
